src/libraries/System.IO.FileSystem/tests/Base/SymbolicLinks/BaseSymbolicLinks.FileSystem.cs

```diff
@@ -0,0 +1,491 @@
+            FileSystemInfo link1TargetInfo = ResolveLinkTarget(link1Path);
+            Assert.True(link1TargetInfo.Exists);
+            AssertIsCorrectTypeAndDirectoryAttribute(link1TargetInfo);
+            Assert.True(link1TargetInfo.Attributes.HasFlag(FileAttributes.ReparsePoint));
+            AssertFullNameEquals(link2Path, link1TargetInfo.FullName);
+            Assert.Equal(link2Target, link1TargetInfo.LinkTarget);
+
+            // link2: do not follow symlinks
+            FileSystemInfo link2TargetInfo = ResolveLinkTarget(link2Path);
+            Assert.True(link2TargetInfo.Exists);
+            AssertIsCorrectTypeAndDirectoryAttribute(link2TargetInfo);
+            Assert.False(link2TargetInfo.Attributes.HasFlag(FileAttributes.ReparsePoint));
+            AssertFullNameEquals(filePath, link2TargetInfo.FullName);
+            Assert.Null(link2TargetInfo.LinkTarget);
+
+            // link1: follow symlinks
+            FileSystemInfo finalTarget = ResolveLinkTarget(link1Path, returnFinalTarget: true);
+            Assert.True(finalTarget.Exists);
+            AssertIsCorrectTypeAndDirectoryAttribute(finalTarget);
+            Assert.False(finalTarget.Attributes.HasFlag(FileAttributes.ReparsePoint));
+            AssertFullNameEquals(filePath, finalTarget.FullName);
+        }
+
+        protected void CreateSymbolicLink_PathToTarget_RelativeToLinkPath_Internal(bool createOpposite)
+        {
+            string tempCwd = GetRandomDirPath();
+            Directory.CreateDirectory(tempCwd);
+            Directory.SetCurrentDirectory(tempCwd);
+
+            // Create a dummy file or directory in cwd.
+            string fileOrDirectoryInCwd = GetRandomFileName();
+            CreateFileOrDirectory(fileOrDirectoryInCwd, createOpposite);
+
+            string oneLevelUpPath = Path.Combine(tempCwd, "one-level-up");
+            Directory.CreateDirectory(oneLevelUpPath);
+            string linkPath = Path.Combine(oneLevelUpPath, GetRandomLinkName());
+
+            // Create a link with a similar Target Path to the one of our dummy file or directory.
+            FileSystemInfo linkInfo = CreateSymbolicLink(linkPath, fileOrDirectoryInCwd);
+            FileSystemInfo targetInfo = linkInfo.ResolveLinkTarget();
+
+            // Verify that Target is resolved and is relative to Link's directory and not to the cwd.
+            Assert.False(targetInfo.Exists);
+            Assert.Equal(Path.GetDirectoryName(linkInfo.FullName), Path.GetDirectoryName(targetInfo.FullName));
+        }
+
+        public static IEnumerable<object[]> ResolveLinkTarget_PathToTarget_Data
+        {
+            get
+            {
+                foreach(string path in PathToTargetData)
+                {
+                    yield return new object[] { path, false };
+                    yield return new object[] { path, true };
+                }
+            }
+        }
+
+        internal static string[] PathToTargetData => new[]
+        {
+#if WINDOWS
```

Comment:
Similar to the above, you should rather use `TARGET_WINDOWS` or even avoid the `#if` at all and use `OperatingSystem.IsWindows()` (that would be my personal preference)